pkg/registry/generic/etcd/etcd.go

```diff
@@ -285,6 +286,14 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 			if err != nil {
 				return nil, nil, err
 			}
+			if newVersion == 0 {
+				// TODO: The Invalid error should has a field for Resource.
+				// After that field is added, we should fill the Resource and
+				// leave the Kind field empty. See the discussion in #18526.
```

Comment:
> Looks fine to me, but I will let @deads2k give the final LGTM.

This is fine for now.  @smarterclayton and I have been going back and forth with each other.  In most places before we were passing back a resource name in the `kind` field, but I made the mismatch painfully obvious.  That has led to a discussion of what `StatusDetails` should actually return to clients and how to populate it, but that discussion is beyond the scope of this pull.

Thanks for calling this out.